app/controllers/public_pages_controller.rb

```diff
@@ -4,6 +4,10 @@ def include_analytics?
     true
   end
 
+  def redirect_locale_home
+    redirect_to root_path, { locale: I18n.locale }
+  end
+
```

Comment:
this will redirect to the homepage after locale has been set based on the logic in application_controller which importantly will inspect the HTTP_ACCEPT_LANGUAGE header and check for the mireembolso domain